shared/actions/fs.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import * as Constants from '../constants/fs'
+import * as FsGen from './fs-gen'
+import * as I from 'immutable'
+import * as RPCTypes from '../constants/types/rpc-gen'
+import * as Saga from '../util/saga'
+import * as Types from '../constants/types/fs'
+import {posix as Path} from 'path'
+
+import type {SagaGenerator} from '../constants/types/saga'
+
+export const fsPathToRpcPathString = (p: Types.Path): string =>
+  Types.pathToString(p).substring('/keybase'.length)
+
+function* folderList(action: FsGen.FolderListLoadPayload): SagaGenerator<any, any> {
+  if (action.payload.path === '/keybase') {
+    yield Saga.cancel()
+    return
+  }
+
+  yield Saga.call(RPCTypes.SimpleFSSimpleFSListRpcPromise, {
+    opID: action.payload.opID,
+    path: {
+      PathType: 1,
+      kbfs: fsPathToRpcPathString(action.payload.path),
+    },
+  })
+
+  yield Saga.call(RPCTypes.SimpleFSSimpleFSWaitRpcPromise, {
+    opID: action.payload.opID,
+  })
+
+  const result = yield Saga.call(RPCTypes.SimpleFSSimpleFSReadListRpcPromise, {
+    opID: action.payload.opID,
+  })
+
+  const rootPath = action.payload.path
```

Comment:
if you're going to pull this out and name it might as well do it first so you can use the name above